packages/pipeline-editor/src/Pipeline.tsx

```diff
@@ -0,0 +1,648 @@
+/*
+ * Copyright 2018-2020 IBM Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { JupyterFrontEnd } from '@jupyterlab/application';
+import { showDialog, Dialog, ReactWidget } from '@jupyterlab/apputils';
+import {
+  DocumentRegistry,
+  ABCWidgetFactory,
+  DocumentWidget
+} from '@jupyterlab/docregistry';
+import { IFileBrowserFactory } from '@jupyterlab/filebrowser';
+import { NotebookPanel } from '@jupyterlab/notebook';
+import { IconRegistry, IIconRegistry } from '@jupyterlab/ui-components';
+
+import { toArray } from '@phosphor/algorithm';
+import { IDragEvent } from '@phosphor/dragdrop';
+import { Widget, PanelLayout } from '@phosphor/widgets';
+
+import {
+  CommonCanvas,
+  CanvasController,
+  CommonProperties
+} from '@elyra/canvas';
+import '@elyra/canvas/dist/common-canvas.min.css';
+import {
+  FrontendServices,
+  NotebookParser,
+  SubmissionHandler
+} from '@elyra/application';
+import 'carbon-components/css/carbon-components.min.css';
+import '../style/index.css';
+
+import * as palette from './palette.json';
+import * as properties from './properties.json';
+import * as i18nData from './en.json';
+import * as React from 'react';
+
+import { IntlProvider } from 'react-intl';
+
+const PIPELINE_ICON_CLASS = 'jp-MaterialIcon elyra-PipelineIcon';
+const PIPELINE_CLASS = 'elyra-PipelineEditor';
+
+const commandIDs = {
+  openPipelineEditor: 'pipeline-editor:open',
+  openDocManager: 'docmanager:open',
+  newDocManager: 'docmanager:new-untitled'
+};
+
+/**
+ * Class for dialog that pops up for pipeline submission
+ */
+export class PipelineSubmissionDialog extends Widget
```

Comment:
I would suggest one widget per file...

Reply:
Though lab includes "helper" widgets like this in the same file, I'm ok with keeping all widgets in separate files for Elyra

Reply:
I'd say for a helper file like this the class can be kept in a file named `dialog.tsx`